lib/p2p/Pool.ts

```diff
@@ -183,27 +183,42 @@ class Pool extends EventEmitter {
    * Attempt to create an outbound connection to a node using its known listening addresses.
    */
   private tryConnectNode = async (node: NodeConnectionInfo, retryConnecting = false) => {
+    if (!await this.tryConnectWithLastAddress(node, retryConnecting)) {
+      await this.tryConnectWithAdvertisedAddresses(node);
```

Comment:
I believe the logic we want here is to first attempt the last address, then the advertised addresses, and then (optionally) retry the last address again. Otherwise we'd immediately go into the long retry loop before trying their other addresses. So we probably want another if statement here:

```typescript
// first try connecting with the last address once, then try connecting with the advertised
// addresses, then if retryConnecting is true keep retrying the last address
if (!await this.tryConnectWithLastAddress(node, false)) {
  if (!await this.tryConnectWithAdvertisedAddresses(node) && retryConnecting) {
    await this.tryConnectWithLastAddress(node, true);
  }
}
```

Reply:
Yes that was my first approach, will do !